d2cli/main.go

```diff
@@ -398,6 +419,19 @@ func compile(ctx context.Context, ms *xmain.State, plugins []d2plugin.Plugin, la
 	}
 	cancel()
 
+	if boardPath != nil {
+		diagram = diagram.GetBoard(*boardPath)
+		if diagram == nil {
+			return nil, false, fmt.Errorf(`board "%s" not found`, *boardPath)
+		}
+		if singleBoard {
+			diagram.Name = ""
```

Comment:
Why erase the name too?

Reply:
If name would not be erased erased, the board would be saved to a directory named after the source file. And writing to stdout would be broken:

```bash
❯ go run main.go --render-target root.x index.d2 -
success: successfully compiled index.d2 to -/x in 71.135ms
```

With name erased everything works as expected.

Reply:
It's outdated now, I removed this piece of code.